tests/Unit/Core/Parser/SyntaxTree/Expression/MathExpressionNodeTest.php

```diff
@@ -39,16 +39,21 @@ public function testEvaluateExpression($expression, array $variables, $expected)
     public function getEvaluateExpressionTestValues()
     {
         return [
-            ['1 gabbagabbahey 1', [], 0],
-            ['1 + 1', [], 2],
-            ['2 - 1', [], 1],
-            ['2 % 4', [], 2],
-            ['2 * 4', [], 8],
-            ['4 / 2', [], 2],
-            ['4 ^ 2', [], 16],
-            ['a + 1', ['a' => 1], 2],
-            ['1 + b', ['b' => 1], 2],
-            ['a + b', ['a' => 1, 'b' => 1], 2],
+            'Invalid operator returns zero' => ['1 gabbagabbahey 1', [], 0],
+            'Simple addition' => ['1 + 1', [], 2],
+            'Chained addition' => ['1 + 1 + 1', [], 3],
+            'Multiplication before addition' => ['1 + 1 * 2', [], 4],
```

Comment:
Uhm, shouldn't this be `3`?

Reply:
The expressions don't know about operator priority, so `1 + 1 = 2 * 2 = 4` is what it sees. It intentionally also doesn't know about parenthesis grouping in order to KISS.

If you know a reasonable way to implement operator priority (see also source of the expression class) I'd definitely consider adding that as a feature.